Src/BootCamp.Chapter/Lesson4.cs

```diff
@@ -0,0 +1,147 @@
+﻿using System;
+
+namespace BootCamp.Chapter
+{
+    class Lesson4
+    {
+        public static void Demo()
+        {
+            Interview();
+            Interview();
+        }
+
+        public static void Interview()
+        {
+            //  gets all the input from the user, if one of them is invalid
+            //  it displays the relevant error code and returns
+            string name = RecursivePromptString("Name: ", "Name cannot be empty");
+            
+            string surname = RecursivePromptString("Surname: ", "Surname cannot be empty");
+            
+            int age = RecursivePromptInt("Age: ");
+
+            float weight = RecursivePromptFloat("Weight(Kg): ");
+
+            float cmHeight = RecursivePromptFloat("Height (cm): ");
+            float mHeight = cmHeight / 100f;
+
+            //  prints the description according to user's input
+            Console.WriteLine($"{name} {surname} is {age} years old, their weight is {weight}Kg and their height is {cmHeight}cm");
+
+            //  calculate BMI and print errormessage if it got an error
+            float bmi = CalculateBmi(weight, mHeight);
+            if (bmi < 0)
+            {
+                Console.WriteLine("Falied to calculate BMI. Reasons:");
+                if (mHeight <= 0)
+                {
+                    Console.WriteLine($"Height cannot be equal or less than zero, but was {cmHeight}cm.");
+                }
+                if (weight <= 0)
+                {
+                    Console.WriteLine($"Weight cannot be equal or less than zero, but was {weight}.");
+                }
```

Comment:
If both cases are true the error message of height has no "equal or" in it! You need to check for that.